tests/components/tplink_omada/test_sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Tests for TP-Link Omada sensor entities."""
+
+from datetime import timedelta
+import json
+from unittest.mock import MagicMock
+
+from syrupy.assertion import SnapshotAssertion
+from tplink_omada_client.definitions import DeviceStatus, DeviceStatusCategory
+from tplink_omada_client.devices import OmadaGatewayPortStatus, OmadaListDevice
+
+from homeassistant.components.tplink_omada.const import DOMAIN
+from homeassistant.components.tplink_omada.coordinator import POLL_DEVICES
+from homeassistant.core import HomeAssistant
+from homeassistant.util.dt import utcnow
+
+from tests.common import MockConfigEntry, async_fire_time_changed, load_fixture
+
+POLL_INTERVAL = timedelta(seconds=POLL_DEVICES + 10)
+
+
+async def test_device_connected_status(
+    hass: HomeAssistant,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test a connection status is reported."""
+    entity = hass.states.get("sensor.test_poe_switch_device_status")
+    assert entity is not None
+    assert entity == snapshot
+
+
+async def test_device_cpu_usage(
+    hass: HomeAssistant,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test a CPU usage is reported correctly."""
+    entity = hass.states.get("sensor.test_router_cpu_usage")
+    assert entity is not None
+    assert entity.state == "16"
+    assert entity == snapshot
+
+
+async def test_device_mem_usage(
+    hass: HomeAssistant,
+    init_integration: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Test a Memory usage is reported correctly."""
+    entity = hass.states.get("sensor.test_poe_switch_memory_usage")
+    assert entity is not None
+    assert entity.state == "20"
+    assert entity == snapshot
```

Comment:
Okay, I do love snapshot tests, but have you considered using the `snapshot_platform` helper? It will snapshot both the state and the entity entry from the registry

Reply:
Hopefully I got the right end of the stick here.

Reply:
not quite yet, but let me explain so you also know what I did (I changed it myself to save you going back to the drawing table).

`snapshot_platform` just iterates over the entity registry and snapshots every entity for that platform for that config entry. It will snapshot both the entity and the state, so you don't need separate tests for the state anymore. This means you can merge these 3 tests into one and just snapshot the whole platform for the data you have put in your mock :)